zlib-rs/src/deflate.rs

```diff
@@ -1298,7 +1296,6 @@ pub(crate) struct State<'a> {
     pub(crate) insert: usize,
 
     pub(crate) w_size: usize,    /* LZ77 window size (32K by default) */
-    pub(crate) w_bits: usize,    /* log2(w_size)  (8..16) */
```

Comment:
my theory for the slowdown is that by removing this field, the following fields also move, and some relevant fields move in/out of a particular cache line. You can test that by just adding a dummy field of the same size (e.g. `_unused: [u8; 8]` and see if that changes anything.